File: examples/android/MSDKSampleAppJava/lib/src/main/java/msdk/java/utils/SecurePreferencesHelper.java
package msdk.java.utils;

import android.content.Context;

import de.adorsys.android.securestoragelibrary.SecurePreferences;

public class SecurePreferencesHelper {

    private static final int chunkSize = 240;

    private static String getNumberOfChunksKey(String key) {
        return key + "_numberOfChunks";
    }

    public static String[] splitEqually(String text, int size) {
        // Give the list the right capacity to start with. You could use an array
        // instead if you wanted.
        String[] ret = new String[((text.length() + size - 1) / size)];

        for (int start = 0, retIndex = 0; start < text.length(); start += size) {
            ret[retIndex] = text.substring(start, Math.min(text.length(), start + size));
            retIndex++;
        }
        return ret;
    }

    public static void setLongStringValue(Context context, String key, String value) {
        //String[] chunks = value.split("(?<=\\G.{240})");
        String[] chunks = splitEqually(value, chunkSize);

        SecurePreferences.setValue(context, getNumberOfChunksKey(key), chunks.length);

        for (int i = 0; i < chunks.length; ++i) {
            SecurePreferences.setValue(context, key + i, chunks[i]);
        }
    }

    public static String getLongStringValue(Context context, String key, String defaultValue) {
        int numberOfChunks = SecurePreferences.getIntValue(context, getNumberOfChunksKey(key), 0);

        if (numberOfChunks == 0) {
            return defaultValue;
        }

        StringBuffer longString = new StringBuffer();
        for (int i = 0; i < numberOfChunks; ++i) {
            longString.append(SecurePreferences.getStringValue(context, key + i, ""));
        }

        return longString.toString();
    }

    public static void removeLongStringValue(Context context, String key) {
        int numberOfChunks = SecurePreferences.getIntValue(context, getNumberOfChunksKey(key), 0);

        //(0 until numberOfChunks).map { SecurePreferences.removeValue("$key$it") }
        for (int i = 0; i < numberOfChunks; ++i) {
            SecurePreferences.removeValue(context, key + i);
        }
        SecurePreferences.removeValue(context, getNumberOfChunksKey(key));
    }

    public static boolean containsLongStringValue(Context context, String key) {
        return SecurePreferences.contains(context, getNumberOfChunksKey(key));
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/utils/SecurePreferencesHelper.kt
package msdk.kotlin.sample.utils

import android.content.Context
import de.adorsys.android.securestoragelibrary.SecurePreferences

object SecurePreferencesHelper {
    private const val chunkSize = 240
    private fun getNumberOfChunksKey(key: String): String {
        return key + "_numberOfChunks"
    }

    fun splitEqually(text: String, size: Int): Array<String?> {
        // Give the list the right capacity to start with. You could use an array
        // instead if you wanted.
        val ret =
            arrayOfNulls<String>((text.length + size - 1) / size)
        var start = 0
        var retIndex = 0
        while (start < text.length) {
            ret[retIndex] = text.substring(start, Math.min(text.length, start + size))
            retIndex++
            start += size
        }
        return ret
    }

    fun setLongStringValue(
        context: Context?,
        key: String,
        value: String
    ) {
        //String[] chunks = value.split("(?<=\\G.{240})");
        val chunks =
            splitEqually(value, chunkSize)
        SecurePreferences.setValue(
            context!!,
            getNumberOfChunksKey(key),
            chunks.size
        )
        for (i in chunks.indices) {
            SecurePreferences.setValue(context, key + i, chunks[i]!!)
        }
    }

    fun getLongStringValue(
        context: Context?,
        key: String
    ): String? {
        val numberOfChunks = SecurePreferences.getIntValue(
            context!!,
            getNumberOfChunksKey(key),
            0
        )
        if (numberOfChunks == 0) {
            return null
        }
        val longString = StringBuffer()
        for (i in 0 until numberOfChunks) {
            longString.append(SecurePreferences.getStringValue(context, key + i, ""))
        }
        return longString.toString()
    }

    fun removeLongStringValue(context: Context?, key: String) {
        val numberOfChunks = SecurePreferences.getIntValue(
            context!!,
            getNumberOfChunksKey(key),
            0
        )

        //(0 until numberOfChunks).map { SecurePreferences.removeValue("$key$it") }
        for (i in 0 until numberOfChunks) {
            SecurePreferences.removeValue(context, key + i)
        }
        SecurePreferences.removeValue(context, getNumberOfChunksKey(key))
    }

    fun containsLongStringValue(
        context: Context?,
        key: String
    ): Boolean {
        return SecurePreferences.contains(
            context!!,
            getNumberOfChunksKey(key)
        )
    }
}